### PR TITLE
Extend `nested_XXX` functions to mappings/dicts.

### PR DESCRIPTION
**Pull request by Guillem96**
Extended `nested_XXX` trainer pt utility functions to work with mappings (dict, OrderedDict, etc.)

Some classes that are modeling the models' outputs, inherit from `ModelOutput` which at turn is an `OrderedDict`. Currently, when applying these `nested_XXX` functions to these models ouputs the code fails with an error.

Extending this nested utilities to work with dictionaries fix this issue.

### CODE REVIEW

File: src/transformers/trainer_pt_utils.py [hunk 1]
```diff
@@ -140,9 +142,13 @@ def find_batch_size(tensors):
 
 
 def nested_numpify(tensors):
-    "Numpify `tensors` (even if it's a nested list/tuple of tensors)."
+    "Numpify `tensors` (even if it's a nested list/tuple/dict of tensors)."
     if isinstance(tensors, (list, tuple)):
         return type(tensors)(nested_numpify(t) for t in tensors)
+
```
Comment (sgugger) [inline src/transformers/trainer_pt_utils.py:148]:
```suggestion
```

File: src/transformers/trainer_pt_utils.py [hunk 2]
```diff
@@ -165,6 +173,10 @@ def nested_xla_mesh_reduce(tensors, name):
 
         if isinstance(tensors, (list, tuple)):
             return type(tensors)(nested_xla_mesh_reduce(t, f"{name}_{i}") for i, t in enumerate(tensors))
+
```
Comment (sgugger) [inline src/transformers/trainer_pt_utils.py:176]:
```suggestion
```

**Review by sgugger — APPROVED** [review]
Left a couple of nits, but this is great! Can you just run `make style` on your branch to fix the quality issue?

**Comment by Guillem96** [comment]
Thank you for your feedback @sgugger . 
Suggestions and style applied!

**Comment by sgugger** [comment]
Thanks a lot!

**Pull request merged**